torch/csrc/autograd/input_metadata.cpp

```diff
@@ -44,7 +46,8 @@ InputMetadata::InputMetadata(const at::Tensor& t)
           t.options(),
           compute_variant_shape(t),
           is_python_dispatch(t),
-          t.is_nested()) {}
+          t.is_nested(),
+          t.grad_dtype()) {}
```

Comment:
Please document that the value of this is frozen during the forward.

Reply:
Ho actually you do update it later? Hopefully that is tested!
Either way, we should document it!